solutions/USACO feb20-clocktree-silver.mdx

```diff
@@ -0,0 +1,108 @@
+---
+id: usaco-1016
+source: USACO Silver 2020 February
+title: Clock Tree
+author: Melody Yu
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_clocktree_silver_feb20.html)
+
+We can build a graph with the rooms as nodes and corridors as edges. Consider a simple case: there are 2 nodes in the graph and one edge.
+Bessie can start from one of the rooms, go to its neighbor, increase the clock, go back, and increase the starting room's clock.
+During this process of moving back and forth, notice that the difference between two rooms' clock readings may only change by 1.
+If the starting clock readings differ by one, Bessie can start from the bigger clock reading room and travel back and forth until both of the clocks point to 12.
+If the starting clock readings differ by more than one, it is impossible for Bessie to make both clocks point to 12.
+
+If there are more nodes, we can treat it similar to 2 nodes by partitioning all the nodes into 2 groups and making this problem a bipartite graph problem. All trees here are biparte graphs, meaning no nodes will have multiple colors. We can partition the nodes into two groups using DFS, assuming that any two neighbor nodes belong to different groups.
```

Comment:
```suggestion
If there are more nodes, we can treat it similar to two node case by considering the two groups in a bipartite partition. Note that all trees are bipartite graphs, so this partition will always be successful.
```

Just a heads up, it's "bipartite," not "biparte."

Reply:
@frodakcin committed, thanks! Could you check it again?